api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -75,6 +78,16 @@ private String convertClusterName(String workspaceId) {
 
   public ResponseEntity<Cluster> createCluster(String workspaceNamespace,
       String workspaceId) {
+
+    // TODO: enforce access level.
```

Comment:
Can you create a cluster in a read-only workspace?

If not... is the thought that you'd have to clone the workspace to run a notebook?

Reply:
I believe we don't want to be able to create a cluster in a read-only workspace.

I am planning on doing a follow up to figure out what permissions should be in each of these API calls. That may not actually be next on my list though. I wanted to start by getting in enforcing registered auth and workspace access at all.